foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/JVMProblemDetector.java

```diff
@@ -0,0 +1,475 @@
+			String unresolvedModule;
+			if(position > 0){
+				unresolvedModule = line.substring(STRING_2.length(), position).trim();
+			}else{
+				unresolvedModule = line.substring(STRING_2.length()).trim();
+			}
+			structure.addModule(unresolvedModule);
+		} else if(line.startsWith(STRING_3)){
+			//if(unresolvedModules.size() > 0){
+				// parse Java name and version
+				int position = line.indexOf(STRING_4);
+				if(position > 0){
+					int endPosition = line.indexOf(")", position+STRING_4.length());
+					
+					String javaName = line.substring(STRING_3.length(), position).trim();
+					String javaVersion;
+					if(endPosition > 0){
+						javaVersion = line.substring(position+STRING_4.length(), endPosition).trim();
+					}else{
+						javaVersion = line.substring(position+STRING_4.length()).trim();
+					}
+					// store unresolved module
+					structure.addRequieredJava(javaName, javaVersion);
+				}
+		}
+
+	}
+	
+	private boolean isAllowedToShow(){
+		//return false;
+		boolean notToShow = PlatformUI.getPreferenceStore().getBoolean(PREFERENCE_NAME);
+		return !notToShow;
+	}
+	
+	private void setAllowedToShow(boolean allowedToShow){
+		PlatformUI.getPreferenceStore().setValue(PREFERENCE_NAME, !allowedToShow);
+	}
+	
+	class LoggingJob extends UIJob{
+		
+		public LoggingJob() {
+			super(JVMProblemDetectorMessages.SHOW_WARNING_DIALOG);
+		}
+
+		@Override
+		public IStatus runInUIThread(IProgressMonitor monitor) {
+			if(!JVMProblemDialog.showing){
+				structure.beforeReport();
+				
+				JVMProblemDialog dialog = new JVMProblemDialog(Display.getDefault().getActiveShell(), structure, javaVersion);
+				dialog.open();
+				setAllowedToShow(dialog.showNextTime());
+			}
+			Status status = new Status(Status.OK, FoundationCorePlugin.PLUGIN_ID, "");
+			return status;
+		}
+	}
+	
+	private void processStatus(IStatus status){
+		Throwable exception = status.getException();
+		String message = status.getMessage();
+		if(message != null){
+			try {
+				queue.put(message);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		}
+		if(exception != null){
+			message = exception.getMessage();
+			if(message != null){
+				try {
+					queue.put(message);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+			}
+			for(IStatus child : status.getChildren()){
+				processStatus(child);
+			}
+		}
+	}
+
+	@Override
+	public void logging(IStatus status, String plugin) {
+		processStatus(status);
+	}
+	
+	public class UnresolvedStructure{
+		private List<UnresolvedModule> modules = new ArrayList<UnresolvedModule>();
+		
+		private List<String> moduleList = new ArrayList<String>();
+		private String moduleName = null;
+		
+		public UnresolvedModule[] getUnresolvedModules(){
+			ArrayList<UnresolvedModule> list = new ArrayList<UnresolvedModule>();
+			for(UnresolvedModule module : modules){
+				if(module.isNeedReport()){
+					list.add(module);
+				}
+			}
+			return list.toArray(new UnresolvedModule[]{});
```

Comment:
Do we really need to convert this list to an array? Why not to return the list itself here?